types/signed_header.go

```diff
@@ -22,7 +25,70 @@ func (sH *SignedHeader) Verify(untrst header.Header) error {
 		// and is a programmer bug
 		panic(fmt.Errorf("%T is not of type %T", untrst, untrstH))
 	}
+	err := untrstH.ValidateBasic()
+	if err != nil {
+		return &header.VerifyError{
+			Reason: err,
+		}
+	}
+	sHHash := sH.Header.Hash()
+	if !bytes.Equal(untrstH.LastHeaderHash[:], sHHash) {
+		return &header.VerifyError{
+			Reason: fmt.Errorf("last header hash %v does not match hash of previous header %v", untrstH.LastHeaderHash[:], sHHash),
+		}
+	}
+	sHLastCommitHash := sH.Commit.GetCommitHash(&untrstH.Header, sH.ProposerAddress)
+	if !bytes.Equal(untrstH.LastCommitHash[:], sHLastCommitHash) {
+		return &header.VerifyError{
+			Reason: fmt.Errorf("last commit hash %v does not match hash of previous header %v", untrstH.LastCommitHash[:], sHHash),
+		}
+	}
 	return sH.Header.Verify(&untrstH.Header)
 }
 
 var _ header.Header = &SignedHeader{}
+
+// ValidateBasic performs basic validation of a signed header.
+func (h *SignedHeader) ValidateBasic() error {
+	err := h.Header.ValidateBasic()
+	if err != nil {
+		return err
+	}
+
+	err = h.Commit.ValidateBasic()
+	if err != nil {
+		return err
+	}
+
+	// Handle Based Rollup case
+	if h.Validators == nil || len(h.Validators.Validators) == 0 {
+		return nil
+	}
+
+	err = h.Validators.ValidateBasic()
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Equal(h.Validators.Hash(), h.AggregatorsHash[:]) {
+		return errors.New("aggregator set hash in signed header and hash of validator set do not match")
+	}
+
+	// Make sure there is exactly one signature
+	if len(h.Commit.Signatures) != 1 {
+		return errors.New("expected exactly one signature")
+	}
+
+	signature := h.Commit.Signatures[0]
+	proposer := h.Validators.GetProposer()
```

Comment:
`h.Validators.GetProposer()` can return `nil`, and this will cause a panic.

Reply:
It technically can never return `nil` because of the check above that returns if `len(h.Validators.Validators) == 0`. Another reason to modify the `ValidateBasic` code to make it more clearer which https://github.com/rollkit/rollkit/issues/1042 should handle.